schema_based_on_diagram.sql

```diff
@@ -0,0 +1,46 @@
+CREATE TABLE medical_histories(
+    id integer GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    admitted_at timestamp,
+    patient_id integer REFERENCES patients(id)
+);
+
+CREATE TABLE patients (
+    id integer GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    name varchar(100),
+    date_of_birth date
+);
+
+CREATE TABLE invoices(
+    id integer GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    total_amount float,
+    generated_at timestamp,
+    payed_at timestamp,
+    medical_history_id INT REFERENCES medical_histories(id)
+);
+
+CREATE TABLE invoice_item(
+    id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    unit_price float,
+    quantity INT,
+    total_price float,
+    invoice_id INT REFERENCES invoices(id)
+    treatment_id INT REFERENCES treatments(id)
+);
+
+CREATE TABLE treatments(
+    id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    type varchar(100),
+    name varchar(100)
+);
+
+CREATE TABLE medical_histories_treatment(
+    id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    medical_histories_id INT REFERENCES medical_histories(id),
+    treatments_id INT REFERENCES treatments(id)
+);
```

Comment:
Great job @gasangw 🤝

- [ ] Kindly remember to create indexes on the `medical_histories_treatmenet` table for the two foreign keys.

Reply:
Thank you @1GORDON. 
- Changes fully implemented